core/src/server/rpc_module.rs

```diff
@@ -740,17 +740,21 @@ impl<Context: Send + Sync + 'static> RpcModule<Context> {
 				MethodCallback::new_subscription(Arc::new(move |id, params, method_sink, conn, claimed| {
 					let sub_id: RpcSubscriptionId = conn.id_provider.next_id();
 
-					let sink = PendingSubscription(Some(InnerPendingSubscription {
-						sink: method_sink.clone(),
+					let sink = SubscriptionSink {
+						inner: method_sink.clone(),
 						close_notify: Some(conn.close_notify),
 						method: notif_method_name,
 						subscribers: subscribers.clone(),
 						uniq_sub: SubscriptionKey { conn_id: conn.conn_id, sub_id },
-						id: id.clone().into_owned(),
-						claimed,
-					}));
+						id: Some(id.clone().into_owned()),
+						unsubscribe: None,
+						_claimed: claimed,
+					};
 
-					callback(params, sink, ctx.clone());
+					// The callback returns an empty `SubscriptionError` for improved API ergonomics.
```

Comment:
I'm not sure I understand this comment - does it usually do but in this case it's "empty" for the API reasons or is it something else? If it's just an explanation for callback returning `SubscriptionResult`, then I think we're better off leaving that note in the doc comment or near the `F` type near the fn sig above

Reply:
yeah, we could probably clarify that a bit this but it's just an internal hack to make it's a bit more ergonomic 
for folks to return an error when trying to "accept" a subscription.

if it fails such the params could not be parsed we just send a message to subscription call itself, so this not propagated to the user just something internal hack to circumvent that.